### PR TITLE
Add github workflows for gomod and goimports

### PR DESCRIPTION
**Pull request by rajansandeep**
Adds workflow and also cleans up the test failures

### CODE REVIEW

File: generic/main.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func main() {
 	}
 
 	if len(genericList.Items) == 0 {
-		fmt.Fprint(os.Stderr, "You need to define the addon that you want the controller to manager. Please create a" +
+		fmt.Fprint(os.Stderr, "You need to define the addon that you want the controller to manager. Please create a"+
```
Comment (njhale) [inline generic/main.go:100]:
```suggestion
		fmt.Fprint(os.Stderr, "You need to define the addon that you want the controller to manage. Please create a"+
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+          stable: true
+
+      - name: Check code
+        uses: actions/checkout@v2
+
+      - run: find . -name go.mod -execdir go mod tidy \;
```
Comment (njhale) [inline .github/workflows/go.yml:43]: would it make sense to codify some of this as top-level make rules so we can run them locally?

**Comment by stealthybox** [comment]
We should consider ignoring vendor dirs in the context of `go fmt`.
@justinsb mentions some of the benefits of keeping the vendor/ in the installer pkg and it would be a conscious decision to remove it that is out of scope for this patch.

File: .github/workflows/go.yml (hunk 2 above)
Comment (justinsb) [reply .github/workflows/go.yml:43]: That's a good idea - I think we can do that as a separate PR though!

File: generic/main.go (hunk 1 above)
Comment (justinsb) [reply generic/main.go:100]: Suggest we do this in a different PR

File: installer/vendor/github.com/json-iterator/go/any.go [hunk 3]
```diff
@@ -3,11 +3,12 @@ package jsoniter
 import (
 	"errors"
 	"fmt"
-	"github.com/modern-go/reflect2"
 	"io"
 	"reflect"
 	"strconv"
 	"unsafe"
+
+	"github.com/modern-go/reflect2"
```
Comment (justinsb) [inline installer/vendor/github.com/json-iterator/go/any.go:11]: We do need to exclude vendor I think.

**Comment by justinsb** [comment]
This LGTM, but we need to exclude vendor.  It might be easiest as @njhale suggests to put the verification steps into a script / makefile target that the github CI then calls; that makes for faster iteration and a better dev experience.

**Comment by justinsb** [comment]
Searching around the web, the easiest way to ignore the vendor directory is likely to do this:

```
goimports -d $(find . -type f -name '*.go' -not -path "*/vendor/*")
```

(the `go list` alternative approach does not work with nested go modules, AFAICT)

WDYT @rajansandeep ?

**Comment by rajansandeep** [comment]
> Searching around the web, the easiest way to ignore the vendor directory is likely to do this:
> 
> ```
> goimports -d $(find . -type f -name '*.go' -not -path "*/vendor/*")
> ```
> 
> (the `go list` alternative approach does not work with nested go modules, AFAICT)
> 
> WDYT @rajansandeep ?

@justinsb Yes, Thank you!
I was on the same track except that the command you suggested didn't work for me somehow... maybe I had a typo...
Anyways, now the vendor dir is excluded from the goimport checks.

**Comment by justinsb** [comment]
Thanks @rajansandeep - looks great!

/approve
/lgtm

**Pull request merged**